collector/processor/decoupleprocessor/README.md

```diff
@@ -0,0 +1,25 @@
+# Decouple Processor
+
+| Status                   |                       |
+| ------------------------ |-----------------------|
+| Stability                | [in development]      |
+| Supported pipeline types | traces, metrics, logs |
+| Distributions            | [extension]           |
+
+This processor decouples the receiver and exporter ends of the pipeline allowing the lambda function to finish before traces/metrics/logs are exported by the collector. The processor is aware of the Lambda lifecycle and will prevent the environment from being frozen or shutdown until any pending traces/metrics/logs have been exported.
```

Comment:
I suggest including a link to the [lifecycle doc](https://docs.aws.amazon.com/lambda/latest/dg/runtimes-extensions-api.html#runtimes-extensions-api-lifecycle) (I assume the `shutdown` hook is what you're taking advantage of here.